test/integration/controller_instance_test.go

```diff
@@ -0,0 +1,1475 @@
+				}
+				ct.catalogClient = catalogClient
+				ct.client = catalogClient.ServicecatalogV1beta1()
+
+			},
+			postCreateInstance: func(ct *controllerTest) {
+				verifyUsernameInLastBrokerAction(ct.t, ct.osbClient, fakeosb.PollLastOperation, "instance-creator")
+			},
+		},
+	}
+	for _, tc := range cases {
+		t.Run(tc.name, func(t *testing.T) {
+			ct := &controllerTest{
+				t:                            t,
+				broker:                       getTestBroker(),
+				instance:                     getTestInstance(),
+				skipVerifyingInstanceSuccess: tc.skipVerifyingInstanceSuccess,
+				skipInstanceDelete:           tc.skipInstanceDelete,
+				setup:                        tc.setupFunc,
+				preDeleteBroker:              tc.preDeleteBroker,
+				preCreateInstance:            tc.preCreateInstance,
+				postCreateInstance:           tc.postCreateInstance,
+			}
+			ct.run(func(ct *controllerTest) {
+				if err := util.WaitForInstanceCondition(ct.client, testNamespace, testInstanceName, tc.verifyCondition); err != nil {
+					t.Fatalf("error waiting for instance condition: %v", err)
+				}
+			})
+		})
+	}
+}
+
+func TestAttemptProbablyShouldDelete(t *testing.T) {
+	ct := &controllerTest{
+		t:      t,
+		broker: getTestBroker(),
+		instance: func() *v1beta1.ServiceInstance {
+			i := getTestInstance()
+			i.Generation = 2
+
+			operationStartTime := metav1.NewTime(time.Now().Add(-1 * time.Hour))
+			i.Status = v1beta1.ServiceInstanceStatus{
+				Conditions: []v1beta1.ServiceInstanceCondition{{
+					Type:               v1beta1.ServiceInstanceConditionReady,
+					Status:             v1beta1.ConditionFalse,
+					Message:            "Deprovisioning",
+					LastTransitionTime: metav1.NewTime(time.Now().Add(-5 * time.Minute)),
+				}},
+				AsyncOpInProgress:    true,
+				OperationStartTime:   &operationStartTime,
+				CurrentOperation:     v1beta1.ServiceInstanceOperationDeprovision,
+				ReconciledGeneration: 1,
+				ExternalProperties: &v1beta1.ServiceInstancePropertiesState{
+					ClusterServicePlanExternalName: testClusterServicePlanName,
+					ClusterServicePlanExternalID:   testPlanExternalID,
+				},
+				DeprovisionStatus: v1beta1.ServiceInstanceDeprovisionStatusRequired,
+			}
+			i.Status.LastOperation = strPtr("test-operation")
+
+			// Set the deleted timestamp to simulate deletion
+			ts := metav1.NewTime(time.Now().Add(-5 * time.Minute))
+			i.DeletionTimestamp = &ts
+
+			return i
+		}(),
+	}
+	ct.run(func(ct *controllerTest) {
+		condition := v1beta1.ServiceInstanceCondition{
+			Type:   v1beta1.ServiceInstanceConditionReady,
+			Status: v1beta1.ConditionTrue,
+			Reason: "ProvisionedSuccessfully",
+		}
+		if err := util.WaitForInstanceCondition(ct.client, testNamespace, testInstanceName, condition); err != nil {
+			t.Fatalf("error waiting for instance condition: %v", err)
+		}
+	})
+}
+
+func TestServiceInstanceWithHTTPStatusCodeErrorOrphanMitigation(t *testing.T) {
```

Comment:
I think this is all covered by `TestCreateServiceInstanceWithProvisionFailure`.

Reply:
Guess I got a little too carried away.